L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc

```diff
@@ -636,6 +715,137 @@ void L1TCorrelatorLayer1Producer::produce(edm::Event &iEvent, const edm::EventSe
   event_.clear();
 }
 
+void L1TCorrelatorLayer1Producer::rawHgcalClusterEncode(ap_uint<256> &cwrd,
+                                                        const l1ct::DetectorSector<ap_uint<256>> &sec,
+                                                        const l1t::HGCalMulticluster &c) const {
+  cwrd = 0;
+
+  // implemented as of interface document (version of 15/11/2025)
+  ap_ufixed<14, 12, AP_RND_CONV, AP_SAT> w_pt = c.pt();
+  // NOTE: We use iPt here for now despite final HGC FW implementation might be different
+  ap_ufixed<14, 12, AP_RND_CONV, AP_SAT> w_empt = c.iPt(l1t::HGCalMulticluster::EnergyInterpretation::EM);
+
+  // NOTE: this number is not consistent with the iPt interpretation nor with hoe.
+  // hoe uses the total cluster em and had energies while eot computed in the showershapes only accounts for a
+  // a max radius from the cluster center. This is the value used by the ID models
+  ap_uint<8> w_emfrac = std::min(round(c.eot() * 256), float(255.));
+
+  // NOTE II: we compute a second eot value to propagate the value of the total hadronic energy as in
+  // the hoe computation
+  float em_frac_tot = c.hOverE() < 0 ? 0. : 1. / (c.hOverE() + 1.);
+  ap_uint<8> w_emfrac_tot = std::min(round(em_frac_tot * 256), float(255.));
+
+  // LSB defintions according to the HGC/Correlator interface document
+  static constexpr float ETAPHI_LSB = M_PI / 720;
+  static constexpr float SIGMAZZ_LSB = 778.098 / (1 << 7);
+  static constexpr float SIGMAPHIPHI_LSB = 0.12822 / (1 << 7);
+  static constexpr float SIGMAETAETA_LSB = 0.148922 / (1 << 5);
+
+  ap_uint<10> w_eta = round(fabs(c.eta()) / ETAPHI_LSB);
+  ap_int<9> w_phi = round(sec.region.localPhi(c.phi()) / ETAPHI_LSB);
+  // FIXME: we keep subtracting an arbitrary number different from the HLGCal FW one
+  ap_ufixed<12, 11, AP_RND_CONV, AP_SAT> w_meanz = fabs(c.zBarycenter()) - 320;  // LSB = 0.5cm
+
+  ap_uint<6> w_showerlenght = c.showerLength();
```

Comment:
`w_showerlength` and https://github.com/cms-sw/cmssw/blob/d29d4100ada0be4b96ae53eb1085eae4bb1ae7cc/L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc#L769